### PR TITLE
Shipment registration

### PR DESCRIPTION
**Pull request by pskramtai**
Currently work in progress.

This PR Contains initial shipment registration interface and implementation. Currently, some issues exist with automatic Shipment ID generation.

Post machine codes and tracking link are not yet implemented.

### CODE REVIEW

**Comment by yjhn** [comment]
To fix conflicts, just accept the newer (main's) `Persistence/Migrations/DatabaseContextModelSnapshot.cs` and this branch's `Persistence/Database/SeedData.cs`

**Comment by EvalVis** [comment]
Drop the old db, the program should start.

**Comment by EvalVis** [comment]
We need to wait for the authentication to merge this branch.

**Comment by EvalVis** [comment]
Since shipment cannot be saved because sender Id is null, let's not do a fake one. Wait until @vleveris creates sign up and login and merges the branch to main. Then you can merge main to this branch and we can merge this branch then it is fully functional.

**Comment by yjhn** [comment]
Blocked by #16.

**Comment by yjhn** [comment]
Also user ID has to be added to the newly created shipment.

File: Services/AccountService.cs
Comment (EvalVis) [inline Services/AccountService.cs:27]: @vleveris I hope its not a problem if these two methods become public static?

**Comment by EvalVis** [comment]
I think it's done. Currently there is no change at frontend since there is no page to display codes and the link. But on register shipment button click the registration happens. It can be seen at /courier link or db.

**Comment by EvalVis** [comment]
I am suggesting to merge this branch, if you see it fit.

**Comment by pskramtai** [comment]
We can merge this, as it has enough functionality. I can make another PR for stylistic changes later.

File: Services/AccountService.cs
Comment (yjhn) [reply Services/AccountService.cs:27]: Well, they are in practice pure functions, so it doesn't matter; there is no way to abuse them.

**Comment by yjhn** [comment]
Rebased to fix the mess with DB migrations. Also applied a few fixes.

File: Pages/Shipment/ShipmentRegistration.razor
Comment (yjhn) [inline Pages/Shipment/ShipmentRegistration.razor]: For some reason this does not clear the fields. Maybe browser autofills them?

**Comment by yjhn** [comment]
The style desperately needs a major overhaul. I'll try to do it tomorrow morning.

File: Pages/Shipment/ShipmentRegistration.razor
Comment (EvalVis) [reply Pages/Shipment/ShipmentRegistration.razor]: Outdated. Resolving, then, I guess.

**Review by EvalVis — APPROVED** [review]
There is no redirect then clicking register shipment. However, the data appears in the db. I guess that OK to merge.

**Comment by EvalVis** [comment]
Last commit will definitely cause conflict. :)

**Pull request merged**
